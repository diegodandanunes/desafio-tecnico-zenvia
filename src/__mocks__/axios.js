export default {
    get: jest.fn(() => Promise.resolve({data: [{char_id: Math.random(), occupation: []}]}))
}