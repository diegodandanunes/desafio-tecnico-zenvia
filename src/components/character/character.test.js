import React from 'react';
import { render, cleanup, waitFor } from '@testing-library/react';

import { Router } from 'react-router-dom';
import { createMemoryHistory } from 'history';

import Character from './';
import fakeData from '../../__mocks__/character-details';
import axios from 'axios';
const history = createMemoryHistory();

afterEach(cleanup);

it("renders", async() => {
    const { asFragment } = render(<Router history={history}><Character /></Router>);

    await waitFor(() => expect(asFragment()).toMatchSnapshot());
});

it('shows character information', async () => {
    axios.get.mockImplementationOnce(() => Promise.resolve(fakeData))

    const { getByTestId } = render(<Router history={history}><Character /></Router>);
    const name = await waitFor(() => getByTestId('character-name'));
    const birthday = await waitFor(() => getByTestId('character-birthday'));
    const nickname = await waitFor(() => getByTestId('character-nickname'));
    const img = await waitFor(() => getByTestId('character-img'));
    
    expect(name.textContent).toMatch(fakeData.data[0].name);
    expect(birthday.textContent).toMatch(fakeData.data[0].birthday);
    expect(nickname.textContent).toMatch(fakeData.data[0].nickname);
    expect(img.src).toMatch(fakeData.data[0].img);
});