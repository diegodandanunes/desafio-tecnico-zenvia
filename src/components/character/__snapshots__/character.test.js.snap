// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<DocumentFragment>
  <div
    class="sc-AxirZ iXyIpx"
  >
    <div
      class="min-h-100 display-flex"
    >
      <div
        class="display-flex flex-column w-100"
      >
        <header>
          <div
            class="sc-AxirZ iXyIpx"
          >
            <div
              class="sc-AxhCb bGeBvo"
            >
              <img
                alt="Logo Diego Nunes"
                src="logo.png"
              />
              <ul>
                <li>
                  Desafio técnico 
                  <span>
                    |
                  </span>
                   
                  <strong>
                    Zenvia
                  </strong>
                </li>
              </ul>
            </div>
          </div>
        </header>
        <div
          class="sc-AxhUy iyORJJ display-flex align-center justify-center flex-grow flex-wrap"
        >
          <div
            class="sc-AxgMl faNyqb"
          >
            <div
              class="lds-dual-ring"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`renders 2`] = `
<DocumentFragment>
  <div
    class="sc-AxirZ iXyIpx"
  >
    <div
      class="min-h-100 display-flex"
    >
      <div
        class="display-flex flex-column w-100"
      >
        <header>
          <div
            class="sc-AxirZ iXyIpx"
          >
            <div
              class="sc-AxhCb bGeBvo"
            >
              <img
                alt="Logo Diego Nunes"
                src="logo.png"
              />
              <ul>
                <li>
                  Desafio técnico 
                  <span>
                    |
                  </span>
                   
                  <strong>
                    Zenvia
                  </strong>
                </li>
              </ul>
            </div>
          </div>
        </header>
        <div
          class="sc-AxhUy iyORJJ display-flex align-center justify-center flex-grow flex-wrap"
        >
          <div>
            <h1
              data-testid="character-name"
            />
            <i>
              ()
            </i>
            <h4
              data-testid="character-birthday"
            >
              Birthday: 
            </h4>
            <h4
              data-testid="character-nickname"
            >
              Nickname: 
            </h4>
            <img
              alt=""
              data-testid="character-img"
            />
            <a
              class="back"
              href="/"
            >
              <h5>
                Voltar
              </h5>
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
